app/src/main/kotlin/com/wire/android/ui/authentication/create/personalaccount/CreatePersonalAccountViewModel.kt

```diff
@@ -10,81 +10,110 @@ import androidx.lifecycle.viewModelScope
 import com.wire.android.navigation.NavigationCommand
 import com.wire.android.navigation.NavigationItem
 import com.wire.android.navigation.NavigationManager
+import com.wire.android.ui.authentication.create.code.CodeViewModel
+import com.wire.android.ui.authentication.create.code.CodeViewState
+import com.wire.android.ui.authentication.create.details.DetailsViewModel
+import com.wire.android.ui.authentication.create.details.DetailsViewState
+import com.wire.android.ui.authentication.create.email.EmailViewModel
+import com.wire.android.ui.authentication.create.email.EmailViewState
+import com.wire.android.ui.authentication.create.overview.OverviewViewModel
 import dagger.hilt.android.lifecycle.HiltViewModel
-import kotlinx.coroutines.delay
 import kotlinx.coroutines.flow.MutableSharedFlow
 import kotlinx.coroutines.launch
 import javax.inject.Inject
 
-@Suppress("MagicNumber")
+@Suppress("MagicNumber", "TooManyFunctions")
 @OptIn(ExperimentalMaterialApi::class)
 @HiltViewModel
 class CreatePersonalAccountViewModel @Inject constructor(
     private val navigationManager: NavigationManager,
-) : ViewModel() {
+) : ViewModel(), OverviewViewModel, EmailViewModel, DetailsViewModel, CodeViewModel {
```

Comment:
I am a bit confused why are we extending those interfaces ?

Reply:
Thanks to that, every screen/step can work more independently and be a part of multiple flows - each interface is a set of methods that are needed for this step's UI to work and provide the required data to the flow's ViewModel which needs to gather data from multiple screens in order to execute requests

Reply:
all righty, I am not the biggest fan of Inheritance within the ViewModel, is there no way implementing it using Composition (not the jetpack one :P) I think the problem with this approach is adding the abstraction using the interfaces and also passing CreatePersonalAccountViewModel  to the stuff where in the end we just use emailState for example and couple of methods within CreatePersonalAccountViewModel. If we used Composition we eliminate the interfaces and also we use what we really need and not pass more than we need. Do not know how hard it would be hard to achieve it with the code we currently have and how much it will improve the code or if it even will improve anything :P just a gut feeling saying it may be worth to think about Composition instead of Inheritance

Reply:
I thought about not using inheritance, but these ViewPagers will never be fully independent and in different flows they will e.g. make different requests or require different data from other ViewPagers, also in “create team” flow there will be an extra step between “details” and “code”. So it's impossible to completely separate them, we would have to have some abstraction built over this anyway. In practice, such ViewModels would only store states, which would be edited by some other element that would perform requests and manage navigation. We could accomplish all this with many `Flow`s, but that could overcomplicate the code, in the current form I feel it's cleaner. Is this okay for you or do you have idea for a specific solution?

Reply:
all right I trust you on this one :P